Add user annotation for FX graph cache key

shengfukevin opened the pull request:
Summary: AI system co-design team requested to add user annotation for FX graph cache key in PyTorch Kineto trace and Execution trace. With this annotation, they can know the FX graph to which the kernels belong.

Test Plan:
buck2 run mode/opt caffe2/test:test_profiler_cuda -- profiler.test_execution_trace.TestExecutionTraceCUDA

Rollback Plan:

Differential Revision: D79019069




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @Lucaskabela

shengfukevin commented:
@pytorchbot label "topic: not user facing"

sraikund16 reviewed (commented):
Can we add a test to make sure the annotation show up? There should already be inductor tests that use profiler

shengfukevin commented:
> Can we add a test to make sure the annotation show up? There should already be inductor tests that use profiler

@sraikund16, I modified test_execution_trace.py to check if "Call CompiledFxGraph" node is in the trace. Please review it. Thanks

shengfukevin commented:
@pytorch/pytorch-bot merge

shengfukevin commented:
@pytorchbot merge this

shengfukevin commented:
@pytorchbot merge

shengfukevin committed:
Add user annotation for FX graph cache key (#159318)

Summary:

AI system co-design team requested to add user annotation for FX graph cache key in PyTorch Kineto trace and Execution trace. With this annotation, they can know the FX graph to which the kernels belong.

Test Plan:
buck2 run mode/opt caffe2/test:test_profiler_cuda -- profiler.test_execution_trace.TestExecutionTraceCUDA

Rollback Plan:

Reviewed By: sraikund16

Differential Revision: D79019069

shengfukevin commented:
@pytorchbot merge
